### DISCUSSION TITLE
macOS VM does not have "Emulated VLAN" option

### DISCUSSION THREAD

**Discussion by wipika** [utmapp/UTM#5067] [UTM Help (macOS)] [answered]
When I try to create a new macOS VM, (Ventura or Big Sur), the only options under "Network" are "_Shared Network_" and "_Bridged_"

Config:
    UTM Version: Version 4.1.5
    macOS Version: macos ventura
    Mac Chip (Intel, M1, ...): M1

<img width="851" alt="Screenshot 2023-02-13 at 12 23 51 PM" src="https://user-images.githubusercontent.com/13740594/218531680-6a0823e2-722e-4ae3-a2d1-fadf456c30ef.png">

**Comment by wipika** [comment]
anyone?

**Comment by osy** [comment, marked as answer]
Yes, there is no Emulated VLAN for macOS virtualization backend

**Reply by wipika** [reply]
thank you -- I was not aware of that.  So is the "Shared Network" option same as NAT?

**Reply by jBRNDnl** [reply]
Yes. It is the same.

**Comment by polka125** [comment]
I managed to get it to work, step-by-step: 
1.  run `ifconfig` on host to identify VPN interface, normally it will be named something like `utun1`, `utun2`. Sorry, I just fed this to an LLM and asked which is VPN. 

2. Set the network mod to `shared`

3. Run the follwoing script to re-route all traffic to the VPN interface on Host: 

```
#!/bin/bash
VPN_INTERFACE="utun4"
UTM_SHARED_SUBNET="192.168.64.0/24"
sudo sysctl -w net.inet.ip.forwarding=1
echo "nat on $VPN_INTERFACE from $UTM_SHARED_SUBNET to any -> ($VPN_INTERFACE)" | sudo pfctl -f -
sudo pfctl -E
```
4. Done 

I'm not entirely sure about all downsides of this approach, but I could reach otherwise unreachable internet from inside the virtual machine
